### ISSUE TITLE
Deformable convolution visualization

### ISSUE THREAD

**Issue by phongnhhn92** [developer0hye/Simple-PyTorch-Deformable-Convolution-v2#8]
Hi,
Thanks for your simple implementation ! I wonder if we can visualize the learned sampling location just like the author of the paper have shown in their paper ? Something like this:
![Screenshot 2021-09-13 at 10 20 07](https://user-images.githubusercontent.com/11288381/133049156-fd5e377b-feba-45a2-8342-e8e90a80ef2b.png)
I think it has something related to your learned offset and modulation mask but I am not sure how to visualize them as the paper.

**Comment by developer0hye** [comment]
@phongnhhn92 
Ah, it's a good point to show the effect of DCN. I just thought about how to implement this visualization algorithm.
Do you need this function?

**Comment by phongnhhn92** [comment]
Hi,
Yes, I do I found this method improve my model and I wanna know why. I don't know if we can do it since we are using the pytorch function.

**Comment by developer0hye** [comment]
@phongnhhn92 
I think we can visualize offsets. Are you in a hurry? I will try to implement this function within one month.

**Comment by phongnhhn92** [comment]
Hi, 
I would love to see it within 2 weeks. I have a deadline to catch here :(

**Comment by developer0hye** [comment]
@phongnhhn92 

Oh my god, okay, I will try it...

 But you shouldn't believe in me.  HAHA

**Comment by phongnhhn92** [comment]
Hehe, I am trying it on my own as well. But it's good to double check with you because I am not even sure if I am gonna do it right haha.

**Comment by developer0hye** [comment]
@phongnhhn92 
Hi, phong
What have you been up to?


I am working on this issue on [this colab link](https://colab.research.google.com/drive/1_Ox0FKpfzA_oOLXmvxY0IjWDSCB_ZTOl?usp=sharing
).

**Comment by phongnhhn92** [comment]
Hi @developer0hye ,
Thanks for the code ! Can u also upload the example_model_state_dict.pth and the testing image 61.png so I can try to run your code ?

**Comment by developer0hye** [comment]
@phongnhhn92 
Check `main` branch.
I upload the code for offset visualization. Actually, I am not sure that I implemented the code correctly. I need your review.

**Comment by phongnhhn92** [comment]
Hi @developer0hye , I think your implementation is correct. Thank you so much for providing this code. I am gonna close this issue now.
